modules/cudf/src/series/bool.ts

```diff
@@ -28,4 +30,106 @@ export class Bool8Series extends NumericSeries<Bool8> {
   get data() {
     return new Uint8ClampedBuffer(this._col.data).subarray(this.offset, this.offset + this.length);
   }
+
+  protected _prepare_scan_series(skipna: boolean) {
+    if (skipna || !this.hasNulls) { return this; }
+
+    const index = Series.sequence({type: new Int32, size: this.length, step: 1, init: 0});
+
+    const first = index.filter(this.isNull()).getValue(0)!;
+    const slice =
+      Series.sequence({type: new Int32, size: this.length - first, step: 1, init: first});
+
+    const copy = this.cast(this.type);
+    const mask = [...index.cast(new Bool8).fill(true).scatter(false, slice)];
+    copy.setNullMask(mask as any);
+
+    return copy;
+  }
+
+  /**
+   * Compute the cumulative max of all values in this Series.
+   *
+   * @param skipna The optional skipna if true drops NA and null values before computing reduction,
+   * else if skipna is false, reduction is computed directly.
+   * @param memoryResource The optional MemoryResource used to allocate the result Series's device
+   *   memory.
+   * @returns The cumulative max of all the values in this Series.
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   * const a = Series.new([true, false, true])
+   *
+   * a.cummax() // {true, true, true}
+   * ```
+   */
+  cummax(skipna = true, memoryResource?: MemoryResource) {
+    const result_series = this._prepare_scan_series(skipna);
+    return Series.new(result_series._col.cummax(memoryResource));
+  }
+
+  /**
+   * Compute the cumulative min of all values in this Series.
+   *
+   * @param skipna The optional skipna if true drops NA and null values before computing reduction,
+   * else if skipna is false, reduction is computed directly.
+   * @param memoryResource The optional MemoryResource used to allocate the result Series's device
+   *   memory.
+   * @returns The cumulative min of all the values in this Series.
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   * const a = Series.new([true, false, true])
+   *
+   * a.cummin() // {true, false, false}
+   * ```
+   */
+  cummin(skipna = true, memoryResource?: MemoryResource) {
+    const result_series = this._prepare_scan_series(skipna);
+    return Series.new(result_series._col.cummin(memoryResource));
+  }
+
+  /**
+   * Compute the cumulative product of all values in this Series.
+   *
+   * @param skipna The optional skipna if true drops NA and null values before computing
+   *   reduction,
+   * else if skipna is false, reduction is computed directly.
+   * @param memoryResource The optional MemoryResource used to allocate the result Series's device
+   *   memory.
+   * @returns The cumulative product of all the values in this Series.
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   * const a = Series.new([true, false, true])
+   *
+   * a.cumprod() // {1n, 1n, 2n}
```

Comment:
Should these results be `[1n, 0n, 0n]`?

Also, extremely nit picky, I think most of our docs use `[` instead of `{`

Reply:
I picked the `{` up from the libcudf docs, I think it's a nice way to indicate the answer without suggesting that the value is a literal JS array, but I don't have strong feelings here

Reply:
That seems reasonable, no hard preference either.

Reply:
@AjayThorve @trxcllnt do either of yall have a strong preference?